src/engine/controls/cuecontrol.cpp

```diff
@@ -22,19 +23,39 @@ static const double CUE_MODE_NUMARK = 3.0;
 static const double CUE_MODE_MIXXX_NO_BLINK = 4.0;
 static const double CUE_MODE_CUP = 5.0;
 
+constexpr double kNoColorControlValue = -1;
+
+namespace {
+
+// Helper function to convert doubles into RgbColor instances (or nullopt)
+inline mixxx::RgbColor::optional_t doubleToRgbColor(double value) {
```

Comment:
"inline" is not really useful any more.  compilers are very smart and will do what they need to to make things fase.  Let the compiler do what it wants, and leave off the extra keyword

Reply:
If this is should be new coding style rule for us then please propose it and add it to our wiki.

I'm not sure if this is really necessary. Our code is full of inline directives, even at places where inlining is performed implicitly by default.

Reply:
> "inline" is not really useful any more. compilers are very smart and will do what they need to to make things fase. Let the compiler do what it wants, and leave off the extra keyword

Is that so? I thought that it's better to tell the compiler to inline stuff explicitly because they sometimes don't do that to keep the binary small.

Reply:
https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method

> Don't add inline just because you think your code will run faster if the compiler inlines it.
> 
>     When will the compiler not know when to make a function/method 'inline'?
> 
> Generally, the compiler will be able to do this better than you. However, the compiler doesn't have the option to inline code if it doesn't have the function definition. In maximally optimized code usually all private methods are inlined whether you ask for it or not.
> 
> As an aside to prevent inlining in GCC, use __attribute__(( noinline )), and in Visual Studio, use __declspec(noinline).

Reply:
Where should I propose the style clarification? The current wiki doesn't say anything about inline specifically

Reply:
this function is not "very small" as indicated in the guideline. I'm not going to insist on removing the keyword but I doubt one could prove that the code is faster because of it.

Reply:
I recommend to only introduce special rules that differ from common recommendations iff there is a strong and valid reason.

This here might be an edge case. In the end it doesn't really matter, because the compiler is allowed to ignore inline keywords.